### PR TITLE
added `remove_named_servers` setting for jupyterhub-idle-culler

### PR DESCRIPTION
**Pull request by stevejpurves**
If you are using TLJH with named servers, it can be a requirement that the server is removed when culled. This PR is to expose the underlying IdleCuller option, allowing it to be set via `tljh-config`.

- [x] Add / update documentation
- [x] Add tests

### CODE REVIEW

**Comment by stevejpurves** [comment]
I noted some issues with the test suite while getting this to go green. It seems like he mock config can leak between tests meaning it was not possible to add a test for `remove_named_servers` independently of `users`. 

Also I added a test case where potentially the existing test case for services could be extended instead -- let me know if I should change that.

**Review by minrk — APPROVED** [review]

**Review by consideRatio — APPROVED** [review]
This PR has merge conflicts that needs to be manually resolved, but this LGTM and can be merged when those are resolved!

Thank you @stevejpurves for your work on this!!! :heart: :tada: :sunflower:

**Pull request closed without merging**
